mate-panel/button-widget.c

```diff
@@ -158,50 +159,67 @@ button_widget_unset_surfaces (ButtonWidget *button)
 static void
 button_widget_reload_surface (ButtonWidget *button)
 {
-    GdkDisplay *display;
-    gint scale;
+	GdkDisplay *display;
+	gint scale;
 	button_widget_unset_surfaces (button);
 
+	button->priv->needs_move = FALSE;
 	if (button->priv->size <= 1 || button->priv->icon_theme == NULL)
 		return;
 
 	if (button->priv->filename != NULL &&
 	    button->priv->filename [0] != '\0') {
 		char *error = NULL;
-
+		/* icons findable in the icon theme can be handled by gtk directly*/
 		display = gdk_display_get_default ();
 		scale = gtk_widget_get_scale_factor (GTK_WIDGET (button));
-		if  (GDK_IS_X11_DISPLAY (display))
+		GtkIconTheme *icon_theme = gtk_icon_theme_get_default();
+		button->priv->surface =
+			gtk_icon_theme_load_surface (icon_theme,
+				button->priv->filename,
+				button->priv->size,
+				scale,
+				NULL,
+				GTK_ICON_LOOKUP_FORCE_SIZE | GTK_ICON_LOOKUP_FORCE_SVG,
+				NULL);
```

Comment:
Indents
```suggestion
			gtk_icon_theme_load_surface (icon_theme,
										button->priv->filename,
										button->priv->size,
										scale,
										NULL,
										GTK_ICON_LOOKUP_FORCE_SIZE | GTK_ICON_LOOKUP_FORCE_SVG,
										NULL);
```

Reply:
Ok, forget this sugestions.
Indents are not correct with code change. But using tabs in github suggestion function is a mess and looks weird.

Reply:
That whole damned file is infested with tabs, and still worse tabs and spaces everywhere. Maybe a separate PR to reformat it to spaces? To get the formatting to where it is on the new code, I had to have Pluma display it with 2 space tabs, 4 space tabs, and 8 space tabs. What a mess leftover from long ago.

I try to avoid changing just a few tabs to spaces when adding new code, because it then looks nasty anytime it is opened with a different tab width